js/data/locale/my/dateformats.json

```diff
@@ -68,88 +68,88 @@
                 "h": "h",
                 "m": "mm",
                 "s": "ss",
-                "ah": "B h",
-                "hm": "B h:mm",
+                "ah": "a h",
+                "hm": "a h:mm",
                 "ms": "mm:ss",
-                "ahm": "hB h:mm",
-                "hms": "B h:mm:ss",
-                "hmz": "z B h:mm",
-                "ahmz": "z hB h:mm",
-                "ahms": "hB h:mm:ss",
-                "hmsz": "z B h:mm:ss",
-                "ahmsz": "z hB h:mm:ss"
+                "ahm": "a h:mm",
+                "hms": "a h:mm:ss",
+                "hmz": "z a h:mm",
+                "ahmz": "z a h:mm",
+                "ahms": "a h:mm:ss",
+                "hmsz": "z a h:mm:ss",
+                "ahmsz": "z a h:mm:ss"
             },
             "24": {
                 "h": "HH",
                 "m": "mm",
                 "s": "ss",
                 "ah": "HH",
-                "hm": "B H:mm",
+                "hm": "H:mm",
                 "ms": "mm:ss",
-                "ahm": "B H:mm",
-                "hms": "B HH:mm:ss",
-                "hmz": "z B H:mm",
-                "ahmz": "z B H:mm",
-                "ahms": "B HH:mm:ss",
-                "hmsz": "z B HH:mm:ss",
-                "ahmsz": "z B HH:mm:ss"
+                "ahm": "H:mm",
```

Comment:
The cldr tool should make sure to remove the "B" from any of the 24-hour formats.